매우 기본적인? 마이페이지 구현

woosukji opened the pull request:
작업 중인 /me 페이지입니다. 느낌 보시라고~

woosukji committed:
Set sass color palette variables

POOL에서 주로 사용할 컬러팔레트를 sass 변수로 등록합니다.
- `color-` prefix가 붙음
- blue, green, sand와, gray & black 컬러
- 각 색의 `-light`와 `-dark` postfix로 밝기 조정 색 추가
- 단, gray 는 light만 존재, black는 밝기 조정 색 없음

woosukji committed:
Rename AccountMyPage component to MyPage

컴포넌트 이름을 짧게 변경합니다.

woosukji committed:
Set page structure for MyPage(/me)

- /me 페이지를 담당하는 컴포넌트 MyPage 를 셋업합니다.
- 페이지의 기본 구조와 하위 컴포넌트가 들어갈 자리를 잡습니다.
- `contents__tab` (포스트/기록 선택하는 셀렉터) 을 추가합니다.

woosukji committed:
Add to MyPage a profile component(MyPageProfile)

- 프로필 컴포넌트를 추가합니다.

somnisomni commented on Web/vue.config.js:
```diff
@@ -5,4 +5,5 @@ module.exports = {
   transpileDependencies: [
     "vuetify",
   ],
+  css: { loaderOptions: { scss: { additionalData: " @import \"./src/styles/variables.scss\"; " } } },
```
`@/src/styles/variables.scss` 파일은 Vuetify 라이브러리에서 이미 선언되어있는 SCSS variables를 override하기 위한 파일입니다. 해당 파일은 Vuetify에서 자동으로 inject하는지라 따로 로더 옵션에 추가할 필요가 있는지 궁금해요.

아 그리고 override할 것이 아닌 SCSS 변수를 선언하고자 한다면 따로 목적이 있는 `variables.scss` 파일에 추가하지 말고 새 파일 작성하는게 나을 것 같은데 어떻게 생각하세요?

woosukji replied:
이건 제가 잘 몰랐네요. 그럼 로더 옵션만 빼고 variables 에 추가해놓은건 그냥 두면 안될까요? 어차피 색깔 말고 달리 쓸 건 없네요.

somnisomni replied:
따로 선언한 변수들 그게 고대로 SCSS 전역에 사용할 수 있는지 몰라서요... 일단 variables.scss는 뷰티파이에서 오버라이딩용으로 사용되는지라 오버라이딩 가능한 변수 외의 변수는 빌드 시 무시되는지 사용될 수 있는지 확인해봐야 될 것 같아요

woosukji replied:
ㅋㅋ dont touch it works 모드로 가는게 나을 것 같기도 하네요. 여유 생기면 시도해 보겠습니당

somnisomni commented on Web/src/styles/variables.scss:
```diff
@@ -1 +1,17 @@
+$color-sand-dark: #DAD2C3;
+$color-gray: #828282;
+$color-gray-light: #E0E0E0;
+$color-black: #333333;
```
파일 마지막에 줄바꿈이 없는거 확인 부탁드려요!

그리고 색상에 관해서는, 우리만의 팔레트를 꼭 써야한다!가 아니면 Vuetify에 내장된 색상들을 사용하는게 좋을 것 같은데 어떻게 생각하시나용 https://vuetifyjs.com/en/styles/colors/

메인 컬러에 대해서는 저희 한 번 의견을 모아봐야될 것 같네요.

woosukji replied:
색상에 관해서는 우리만의 팔레트를 꼭 쓰는게 나을 것 같아요. 
어차피 내장 색상 쓰려고 해도 어느 것들 쓸지 부분집합을 합의해서 써놔야 하니까 이렇게 등록해 놓는거랑 차이가 없어 보이구요. 
지금 써놓은 색상은 임의로 만든게 아니라 제가 아이템 컨셉 주장할 때 보여드린 색깔 + 프로토 만들면서 필요할 것 같은거 추가한 건데요, 저는 이 팔레트가 괜찮은데 색상 변경안 있으시면 스레드 남겨 주세요.

somnisomni commented on Web/src/pages/account/MyPageRecords.vue:
```diff
@@ -0,0 +1,13 @@
+import Component from "vue-class-component";
+
+@Component
+export default class ProfileAndSettingsPage extends Vue {
```
요것도 클래스명-파일명 일치 부탁드려요.

somnisomni commented on Web/src/pages/account/MyPagePosts.vue:
```diff
@@ -0,0 +1,13 @@
+import Component from "vue-class-component";
+
+@Component
+export default class ProfileAndSettingsPage extends Vue {
```
컴포넌트로 사용될 계획인거죠? 가급적 클래스명과 파일명 일치시키는걸 추천드리고 페이지가 아닌 일반 컴포넌트는 명확한 구분을 위해 `@/components`에 모아두는건 어떻게 생각하시나요? 저는 일단 그렇게 작업하고 있긴 한데

woosukji replied:
여기 들어간 컴포넌트가 여러 군데 재사용 할 일이 아니라, 한 페이지에 코드가 너무 길어지니까 읽는데 방해돼서 블럭별로 나눈 거거든요. 
지금 /me 페이지에 쓰이는 컴포넌트만 대여섯 개인데, 이런식으로 다 components 파일에 모아두는 게 더 헷갈릴 것 같아요. 
그 안에서 hierarchy를 나누면 되겠지만, 그럼 사실 지금 저처럼 각 페이지 디렉토리에 쓰임별로 모아두는 거랑 다를 게 없어 보이구요.
페이지를 담당하는 컴포넌트는 `*page.vue`, 안에 들어가는 (페이지 종속적인) 자식 컴포넌트는 페이지 이름에 이어서 `*page**.vue` 식으로 이름 붙여놔서 보는데 혼란스럽지 않을 것 같아요.
민규님이 쓰신 포스트 카드는 여기저기서 다 쓸거기 때문에 거기 두는 게 맞습니다.

woosukji replied:
/me 에서 쓰는 커다란 프로필 컴포넌트는 다시 쓸 것 같아서 components로 옮겨놓겠습니다.

wooksong replied:
제가 Web 개발 경력이 일천하고, 현재 코드가 WiP인 것으로 보여 맞는 말씀을 드릴 수 있을지 모르겠습니다만,

> 여기 들어간 컴포넌트가 여러 군데 재사용 할 일이 아니라, 한 페이지에 코드가 너무 길어지니까 읽는데 방해돼서 블럭별로 나눈 거거든요.

컴포넌트는 재사용 여부와 상관 없이, ```@/components``` 밑에 있는 것이 **원론적으로** 맞다고 알고 있습니다. 근데, @woosukji 님이 말씀하신 것과 같이 (말씀하셨나요?) routing을 위한 component들은 관련이 있는 ```@/views```, ```@/pages```에 넣습니다. vue-cli가 이렇게 하기도 하고, 누군가는 ``` <router-view>``` component니깐 ```@/views```, ```@/pages```로 봐야한다고 하기도 하고...

보통 이런 것에는 답이 없고, 합의된 것으로 하면 된다고 생각하는데 저는 @woosukji 님의 방식을 더 선호합니다. (뭐 중요하진 않습니다. 전 초보 웹 개발자라서요...)
있는 것을 더 선호할 수 있습니다. 왜냐면 얘들은 component인데 Page/View의 역할을 하는 것이자나요? 

이걸 정확히 설명한 글은 못 찾았고, 대충 비슷한 stackoverflow 답변을 붙여요. FYI, https://stackoverflow.com/questions/50865828/what-is-the-difference-between-the-views-and-components-folders-in-a-vue-project

> 지금 /me 페이지에 쓰이는 컴포넌트만 대여섯 개인데, 이런식으로 다 components 파일에 모아두는 게 더 헷갈릴 것 같아요.
그 안에서 hierarchy를 나누면 되겠지만, 

vue 공식 가이드는 이러지 말라고 하는데, 이건 arguing이 좀 있는 영역이더라구요. 일단 vue에서 가이드하는 네이밍이 파일이 한 두개일때는 모르겠는데, 많아지면 뭐가 뭔지도 모르겠고, 그걸 플랫하게 두는게 장점이 뭔지도 모르겠고 -_-; 그렇다고 https://vueschool.io/articles/vuejs-tutorials/structuring-vue-components/ 이렇게 하는 것도 저는 지저분해서 별로라 생각하고...

애초에 component를 잘 설계하시고 [mixin](https://kr.vuejs.org/v2/guide/mixins.html)이나 [async component](https://vuejs.org/v2/guide/components-dynamic-async.html) 쓸 수 있는 곳에는 최대한 써서 가독성을 높이는 방법 밖에는 없는 것 같네요. 

재밌는 주제라 저도 모르게 댓글을 달아보았습니다!

somnisomni reviewed (changes requested):
😄

woosukji committed:
Refactor styles for MyPage components

- `.profile-container` 을 제거해서 마크업을 flatten 합니다.
- 컴포넌트 루트에 class와 style이 붙던 것을 부모 컴포넌트로 옮겨 옵니다.

woosukji committed:
Add to MyPage a credit component(MyPageCredits)

- 가감점 컴포넌트 레이아웃 만들어추가합니다.

woosukji committed:
Add to MyPageCredits an input form

woosukji committed:
Redesign MyPageCredits component

- 가감점 카드 안의 요소들을 다시 배치합니다.
- 특히 timeline의 각 item 안에서의 layout을 대폭 수정했습니다.
- timeline 디자인 조정을 위해 전역 변수를 등록합니다.

woosukji committed:
Redesign positionings for MyPage

woosukji committed:
Move manual style variables to dedicated new file

- #11 에서 논의한 색상 변수를 따로 파일을 마련해서 옮깁니다.

woosukji committed:
Correct component exportation names

woosukji committed:
Adjust positions in MyPage: profiles & tab background

- github 프로필 페이지와 같이 tab bar 뒤로 사라지는 프로필을 구현합니다.
- MyPageAwards는 Credits의 복사판으로, 이후 refactoring 커밋에서 사라질 예정입니다.

woosukji committed:
Add interfaces for records

- 공적기록 기능에 사용되는 인터페이스를 추가합니다.
- 이후 커밋에서 민규님의 @/interfaces (#13) 으로 이동할 예정입니다.

woosukji committed:
Refactor MyPage credits & awards into single component

- MyPageCredits 컴포넌트의 내용이 MyPageAwards(예정) 에서도 다시 쓰이므로 이를 한 컴포넌트로 통일합니다.
- data와 props를 연결하고, create 와 delete 로직을 만듭니다.
- update 로직도 시도했으나 ref 관련 문제로 잘 되지 않아 미포함합니다.

woosukji committed:
Add to MyPage basic event handlers for MyPageRecords

이전 커밋의 MyPageRecords 변경사항에 맞춰 컴포넌트를 사용하고 더미 데이터를 추가합니다.

woosukji committed:
Resolve conflicts from rebasing

woosukji commented:
- 현재 포스트 탭이 비어있는데, 이건 민규님 피드 구현 (#12 ) 를 이어받아 추가하겠습니다.
- 모바일 대응은 다음에 하겠습니다.

somnisomni reviewed (dismissed):
고생하셨서요

크게 문제될 코드는 없는 것 같구용, 대신 코딩 컨벤션에 대해서도 이야기 나눌 수 있다면 좋을텐데 지금 그럴 시간이 어딨겠어요 껄껄🤣

somnisomni committed:
Style quick fix

- 메인 컨테이너의 패딩을 제거합니다.
- `MyPage.vue`의 탭 아이템에 더 넉넉한 상단 마진을 줍니다.
